profile/profile_worker.go

```diff
@@ -123,3 +106,27 @@ func collectSample(pc []uintptr) {
 	}
 
 }
+
+func filterSCSPrivateFunc(f string) bool {
+	const scsPrefix = "github.com/consensys/gnark/frontend/cs/scs.(*builder)."
+	if idx := strings.Index(f, scsPrefix); idx != -1 && len(f) > (idx+len(scsPrefix)) {
```

Comment:
Is there any case where idx isn't 0 or 1? Isn't the question essentially "does f start with scsPrefix". Can there be anything before? Maybe for readibility could then instead use `strings.HasPrefix`?